include/z64cutscene.h

```diff
@@ -3,286 +3,480 @@
-    /* 0x04 */ u16 endFrame;
+    /* 0x00 */ u8 unused_00[0x2];
+    /* 0x02 */ u16 frame;
+    /* 0x04 */ u8 unused_04[0x2];
     /* 0x06 */ u8  hour;
     /* 0x07 */ u8  minute;
-} CsCmdDayTime; // size = 0x8
+    /* 0x08 */ u8 unused_08[0x4];
+} CsCmdTime; // size = 0xC
 
 typedef struct {
-    /* 0x00 */ u16 base;
+    /* 0x00 */ u16 destination;
+    /* 0x02 */ u16 frame;
+    /* 0x04 */ u8 unused_04[0x4];
+} CsCmdDestination; // size = 0x8
+
+typedef struct {
+    /* 0x00 */ u16 textId; // can also be an ocarina action for `CS_TEXT_OCARINA_ACTION`
     /* 0x02 */ u16 startFrame;
     /* 0x04 */ u16 endFrame;
     /* 0x06 */ u16 type;
-    /* 0x08 */ u16 textId1;
-    /* 0x0A */ u16 textId2;
-} CsCmdTextbox; // size = 0xC
+    /* 0x08 */ u16 altTextId1;
+    /* 0x0A */ u16 altTextId2;
+} CsCmdText; // size = 0xC
+
+#define CS_TEXT_ID_NONE 0xFFFF
+
+typedef struct {
+    /* 0x00 */ u16 type;
+    /* 0x02 */ u16 startFrame;
+    /* 0x04 */ u16 endFrame;
+    /* 0x06 */ u8 unused_06[0x2];
+} CsCmdTransition; // size = 0x8
 
 typedef struct {
-    /* 0x00 */ u16 action; // "dousa"
+    /* 0x00 */ u16 id; // "dousa"
     /* 0x02 */ u16 startFrame;
     /* 0x04 */ u16 endFrame;
-    union {
-        /* 0x06 */ Vec3s rot;
-        /* 0x06 */ Vec3us urot;
-    };
+    /* 0x06 */ Vec3us rot;
     /* 0x0C */ Vec3i startPos;
     /* 0x18 */ Vec3i endPos;
-    /* 0x24 */ Vec3i normal;
-} CsCmdActorAction; // size = 0x30
+    /* 0x24 */ u8 unused_24[0xC];
+} CsCmdActorCue; // size = 0x30
 
-typedef enum {
-    CS_STATE_IDLE,
-    CS_STATE_SKIPPABLE_INIT,
-    CS_STATE_SKIPPABLE_EXEC,
-    CS_STATE_UNSKIPPABLE_INIT,
-    CS_STATE_UNSKIPPABLE_EXEC
-} CutsceneState;
-
-typedef enum {
-    CS_CMD_00 = 0x0000,
-    CS_CMD_CAM_EYE = 0x0001,
-    CS_CMD_CAM_AT = 0x0002,
-    CS_CMD_MISC = 0x0003,
-    CS_CMD_SET_LIGHTING = 0x0004,
-    CS_CMD_CAM_EYE_REL_TO_PLAYER = 0x0005,
-    CS_CMD_CAM_AT_REL_TO_PLAYER = 0x0006,
-    CS_CMD_07 = 0x0007,
-    CS_CMD_08 = 0x0008,
-    CS_CMD_09 = 0x0009,
-    CS_CMD_TEXTBOX = 0x0013,
-    CS_CMD_SET_PLAYER_ACTION = 0x000A,
-    CS_CMD_SET_ACTOR_ACTION_1 = 0x000F,
-    CS_CMD_SET_ACTOR_ACTION_2 = 0x000E,
-    CS_CMD_SET_ACTOR_ACTION_3 = 0x0019,
-    CS_CMD_SET_ACTOR_ACTION_4 = 0x001D,
-    CS_CMD_SET_ACTOR_ACTION_5 = 0x001E,
-    CS_CMD_SET_ACTOR_ACTION_6 = 0x002C,
-    CS_CMD_SET_ACTOR_ACTION_7 = 0x001F,
-    CS_CMD_SET_ACTOR_ACTION_8 = 0x0031,
-    CS_CMD_SET_ACTOR_ACTION_9 = 0x003E,
-    CS_CMD_SET_ACTOR_ACTION_10 = 0x008F,
-    CS_CMD_SCENE_TRANS_FX = 0x002D,
-    CS_CMD_NOP = 0x000B,
-    CS_CMD_PLAYBGM = 0x0056,
-    CS_CMD_STOPBGM = 0x0057,
-    CS_CMD_FADEBGM = 0x007C,
-    CS_CMD_SETTIME = 0x008C,
-    CS_CMD_TERMINATOR = 0x03E8,
-    CS_CMD_END = 0xFFFF
-} CutsceneCmd;
+typedef struct {
+    /* 0x00 */ s8 continueFlag;
+    /* 0x01 */ s8 cameraRoll;
+    /* 0x02 */ u16 nextPointFrame;
+    /* 0x04 */ f32 viewAngle; // in degrees
+    /* 0x08 */ Vec3s pos;
+} CutsceneCameraPoint; // size = 0x10
```

Comment:
the padding should probably be explicit like elsewhere to match the `CS_CAM_POINT` macro